### ISSUE TITLE
sync not honoring s2s-preserve-access-tier

### ISSUE THREAD

**Issue by mrhoads** [Azure/azure-storage-azcopy#2208]
### Which version of the AzCopy was used? 
10.18.1
### Which platform are you using? (ex: Windows, Mac, Linux)
MacOS 12.6.5

### What command did you run?

`azcopy sync "https://hotdefaultaccount.blob.core.windows.net/source/" "https://cooldefaultaccount.blob.core.windows.net/target/"  --s2s-preserve-access-tier=false`

### What problem was encountered?
I expect setting *s2s-preserve-access-tier=false* to result in the sync in the destination using the default access tier set for the storage account.  The source has a default access tier of hot, while the destination has a default access tier of cool.

### How can we reproduce the problem in the simplest way?
- create two storage accounts
- set the default access tier of one account to hot
- set the default access tier of the other to cool
- upload one or more blobs to a container in the hot account
- run `azcopy sync "<hot source account>/container/" "<cool destination account>/container/" --s2s-preserve-access-tier=false`
- verify that the destination account blob(s) have an access tier of hot

### Have you found a mitigation/solution?
Blobs could use the Azure CLI, PowerShell, or other means to set the access tier to cool after syncing.  This scenario is to support a manual BC/DR operation in which it's desirable to set the access tier of all synced blobs to a cheaper tier.  It looks like azcopy copy [already](https://learn.microsoft.com/en-us/azure/storage/blobs/access-tiers-online-manage?tabs=azcopy#copy-a-blob-to-a-different-online-tier) implements this.  I may be mistaken on the purpose of the s2s-preserve-access tier flag, but ideally, it or another method would allow azcopy sync to use the destination account's default access tier.

**Comment by siminsavani-msft** [comment]
Hi @mrhoads ! Just wanted to clarify a few things. Your expectation is that after blobs are downloaded to the destination (cool tier), the blobs should also have the cool tier, is this correct? Or are you expecting that the blobs should be of the hot tier. 

Setting `s2sPreserveAccessTier=false` would result in the access tier not copying over which means the blobs with get the destination storage account's default access tier (in this case, the blobs would be of cool tier). 

If you are wanting preserve the tier of the source blobs, I would recommend setting `s2sPreserveAccessTier=true`.

**Comment by mrhoads** [comment]
Yes, @siminsavani-msft, I'd expect that if the destination has a default access tier of cool, using `s2sPreserveAccessTier=false` will set the synced blobs to cool but in my testing on version 10.18.1, they're instead using the access tier of the source storage account (hot in my case).

**Comment by siminsavani-msft** [comment]
No worries! Glad to hear it worked out in the end @mrhoads !
